### PR TITLE
Add Quota implementation

### PR DESCRIPTION
**Pull request by hardillb**
```yaml
driver:
  type: s3
  quota: [number of bytes to allow per project]
  options:
    ...
```

```yaml
context:
  type: sequelize
  quota: [number of bytes to allow per project
  options:
    ...
```

If quota is not set then no limits will be enforced

Moves PostgreSQL to sequelize to make it fit better.

This also automatically creates the PG database if it's not found.

Finally passed the whole `app` object to the drivers so they can make use of logging

### CODE REVIEW

**Comment by hardillb** [comment]
Over quota failures use HTTP Statue 413 codes

File: README.md [hunk 1]
```diff
@@ -168,37 +168,37 @@ context:
...
+    username: user
+    password: password
 ```
 
+To use with SQLite configure as follows:
+
 ```yaml
 context:
-  type: postgres
+  type: squelize
```
Comment (Steve-Mcl) [inline README.md:193]:
```suggestion
  type: sequelize
```

File: forge/context-driver/redis.js [hunk 2]
```diff
@@ -92,17 +92,20 @@ async function recursiveInsert (projectId, scope, path, value) {
 }
 
 module.exports = {
-    init: async function (options) {
+    init: async function (app) {
+        const url = app.config.context.options.url || 'redis://localhost:6379'
```
Comment (Steve-Mcl) [inline forge/context-driver/redis.js:96]:
```suggestion
        const url = app.config.context.options?.url || 'redis://localhost:6379'
```

**Review by Steve-Mcl — APPROVED** [review]
optional suggestions :)

**Pull request merged**